modules/javafx.controls/src/test/java/test/javafx/scene/control/TableViewHorizontalArrowsTest.java

```diff
@@ -0,0 +1,226 @@
+
+import javafx.geometry.NodeOrientation;
+import javafx.scene.control.SelectionMode;
+import javafx.scene.control.TableColumn;
+import javafx.scene.control.TablePosition;
+import javafx.scene.control.TableView;
+import test.com.sun.javafx.scene.control.infrastructure.KeyEventFirer;
+import test.com.sun.javafx.scene.control.infrastructure.KeyModifier;
+import test.com.sun.javafx.scene.control.infrastructure.StageLoader;
+
+/**
+ * Test basic horizontal navigation mappings for TableView. It's parametrized on NodeOrientation
+ */
+@RunWith(Parameterized.class)
+public class TableViewHorizontalArrowsTest {
+    @Parameterized.Parameters
+    public static Collection<?> implementations() {
+        return Arrays.asList(new Object[][] {
+            {NodeOrientation.LEFT_TO_RIGHT},
+            {NodeOrientation.RIGHT_TO_LEFT}
+        });
+    }
+
+    private TableView<String> tableView;
+    private TableView.TableViewSelectionModel<String> sm;
+    private TableView.TableViewFocusModel<String> fm;
+
+    private TableColumn<String, String> col0;
+    private TableColumn<String, String> col1;
+    private TableColumn<String, String> col2;
+    private TableColumn<String, String> col3;
+    private TableColumn<String, String> col4;
+
+    private KeyEventFirer keyboard;
+    private StageLoader stageLoader;
+    private NodeOrientation orientation;
+
+    public TableViewHorizontalArrowsTest(NodeOrientation val) {
+        orientation = val;
+    }
+
+    @Before
+    public void setup() {
+        tableView = new TableView<String>();
+        tableView.setNodeOrientation(orientation);
+        sm = tableView.getSelectionModel();
+        fm = tableView.getFocusModel();
+
+        sm.setSelectionMode(SelectionMode.MULTIPLE);
+        sm.setCellSelectionEnabled(false);
+
+        tableView.getItems().setAll("1", "2", "3", "4", "5", "6", "7", "8", "9",
+                                    "10", "11", "12");
+
+        col0 = new TableColumn<String, String>("col0");
+        col1 = new TableColumn<String, String>("col1");
+        col2 = new TableColumn<String, String>("col2");
+        col3 = new TableColumn<String, String>("col3");
+        col4 = new TableColumn<String, String>("col4");
+        tableView.getColumns().setAll(col0, col1, col2, col3, col4);
+
+        keyboard = new KeyEventFirer(tableView);
+
+        stageLoader = new StageLoader(tableView);
+        stageLoader.getStage().show();
+    }
+
+    @After
+    public void tearDown() {
+        tableView.getSkin().dispose();
+        stageLoader.dispose();
+    }
+
+    // ---------------- Helper methods -------------------------
+    /**
+     * Toggles the nodeOrientation of tableView.
+     */
+    private void changeNodeOrientation() {
+        orientation = (orientation == NodeOrientation.LEFT_TO_RIGHT?
+            NodeOrientation.RIGHT_TO_LEFT : NodeOrientation.LEFT_TO_RIGHT);
+        tableView.setNodeOrientation(orientation);
+    }
+
```

Comment:
changing the parameter of the test is at least unusual, should be doc'ed so future contributors are aware of it

Reply:
Good catch!!!
It definitely would have unintentional side effect of incorrect orientation in subsequent tests.
I updated the test to restore the orientation at the end of the test. Also documented the same.

Reply:
hmm ... don't think that it will effect subsequent tests, though might be mis-understanding the api doc of Parameterized:

> When running a parameterized test class, instances are created for the cross-product of the test methods and the test data elements.

So just suggested to change the doc of the method, something like

    Changes the parameter nodeOrientation and sets the table's orientation to the new value

Alternatively, leave the parameter alone, just toggle the table's orientation and change the forward/backward methods to choose the keys based on the table's current orientation. Or fully parameterize on the keys - sry, can't resist :)

Reply:
Thanks for clarifying my impression about side effect on subsequent tests.
I have the updated the method documentation (also changed the method name to toggleNodeOrientation() to clearly reflect what it does)

There would always be different ways to test the same thing. I think, current test is in good shape and suffices the purpose of unit testing this fix.